clappr/clappr-ios
# Sends cookies associated with a domain in AVURLAsset for tvOS

## Goal

To send cookies associated with a domain when instantiating an `AVURLAsset`.

## How to test

Replace `viewDidLoad()` in `Example/Clapper_tvOS/ViewController.swift` code for this:

```swift
override func viewDidLoad() {
        super.viewDidLoad()

        let cookie = HTTPCookie(properties: [HTTPCookiePropertyKey.domain: "clappr.io",
                                             HTTPCookiePropertyKey.path: "/",
                                             HTTPCookiePropertyKey.name: "testing",
                                             HTTPCookiePropertyKey.value: "it was sent"])!
        HTTPCookieStorage.shared.setCookie(cookie)

        let options = [
            kSourceUrl: "http://clappr.io/highline.mp4",
            kAutoPlay: true,
            kMediaControl: true
            ] as [String: Any]
        player = Player(options: options)

        listenToPlayerEvents()

        addChildViewController(player)
        player.view.frame = view.bounds
        view.addSubview(player.view)
        player.didMove(toParentViewController: self)
    }
```
This code will set a cookie for the domain clappr.io. Then, play the video and check that the cookie was sent (using Charles, for example):

<img width="1679" alt="screen shot 2018-05-24 at 14 02 42" src="https://user-images.githubusercontent.com/8026597/40502561-b35af510-5f61-11e8-8ced-bcd44554dd3e.png">
